### ISSUE TITLE
Facebook and Twitter login fail if [Parse setApplicationId:clientKey:] is called multiple times

### ISSUE THREAD

**Issue by jimspeth** [ParsePlatform/Parse-SDK-iOS-OSX#625]
Our application has a legitimate use for changing Parse apps while running.  In Parse 1.8, this worked without a problem.

However, when we tried upgrading to Parse 1.10, Facebook and Twitter login is failing.  The call to `[PFFacebookUtils logInInBackgroundWithReadPermissions:block:]`is called back with `user` and `error` both nil.  Internally, there is an exception that is caught but not propagated:

```
No registered authentication delegate found for `facebook` authentication type.
Register a delegate first via PFUser.registerAuthenticationDelegate(delegate, forAuthType:)
```

I see in https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/issues/159 that apparently our use case is not supported.  Please consider making this a supported use case and fixing this issue.

**Comment by nlutsenko** [comment]
Hey @jimspeth, unfortunately the behavior is `by design` undefined when calling `Parse.+setApplicationId:clientKey:` multiple times in a single application run.

I will mark this issue as discussion/enhancement, as I can see small narrow set of apps that could benefit from it and we might support it later.

**Comment by jimspeth** [comment]
@nlutsenko How about just exposing a `+[Parse reset]` method that just resets all state?  It seems like this would be fairly straight-forward.

I've been able to work around this in our code by calling these private methods:

```
    [Parse _clearCurrentManager];
    [PFAnonymousUtils _clearAuthenticationProvider];
    [PFFacebookUtils _setAuthenticationProvider:nil];
    [PFTwitterUtils _setAuthenticationProvider:nil];
```

Anything I'm missing resetting?

**Comment by nlutsenko** [comment]
Implemented in #874
